aea/protocols/base.py

```diff
@@ -309,21 +309,13 @@ def from_config(cls, configuration: ProtocolConfig) -> "Protocol":
         assert (
             configuration.directory is not None
         ), "Configuration must be associated with a directory."
-        directory = configuration.directory
-        package_modules = load_all_modules(
-            directory, glob="__init__.py", prefix=configuration.prefix_import_path
+        load_aea_package(configuration)
+        class_module = importlib.import_module(
```

Comment:
this is important: `importlib.import_module` will check the `sys.modules` cache, which has been populated in the line above with `load_aea_package`.

Notice that it is different from the previous approach. We re-loaded the module, hence we lose the changes done in other modules (e.g. in __init__.py when we set the `message.serialzier` field.)

Reply:
`load_aea_package` will guarantee that in `sys.modules` there are the modules we want, and not other modules loaded earlier. 

However, that might cause previously loaded modules to not work correctly. It is a known issue.